### ISSUE TITLE
Using angle bracket type assertions causes parsing error

### ISSUE THREAD

**Issue by mischah** [nzakas/eslint-plugin-typescript#130]
Hej, I love this plugin and prefer using this with [TypeScript ESLint Parser](https://github.com/eslint/typescript-eslint-parser) over TSLint.

I only have the following error when using angle bracket type assertions.

## Current behavior

ESLint reports the following:

```shell
/projectRoot/src/testCase.ts
  10:9  error  Parsing error: '}' expected
```

## Expected behavior

According to the rule [no-angle-bracket-type-assertion](https://github.com/nzakas/eslint-plugin-typescript/blob/master/docs/rules/no-angle-bracket-type-assertion.md) I expected something like

```shell
/projectRoot/src/testCase.ts
  10:9  error  Prefer 'as View' instead of '<View>' when doing type assertions
```

## How to reproduce

```typescript
class View {
  id:string;

  name:string;
}

const base = { id: '1'};
const another = { name : 'View 1'};
const view = <View>{
  ...base,
  ...another
};

```

cc// @nicolaisueper

**Comment by macklinu** [comment]
I believe this is an issue with [eslint/typescript-eslint-parser](https://github.com/eslint/typescript-eslint-parser) that should be tracked in that repo as well, but could someone with more expertise/understanding of the typescript-eslint-parser confirm that please? 

Here's a reproducible [AST explorer](http://astexplorer.net/#/gist/58537ff3964ce6aff61922af7e721663/c65480857ca3bf6c4e5aa8bd7274d0e8c5379834) as well.
